### PR TITLE
TDQ-13704 TDQ-13979 make run with sample data checkbox work friendly

### PR DESCRIPTION
**Pull request by zshen-talend**

### CODE REVIEW

File: main/plugins/org.talend.dataprofiler.core/src/org/talend/dataprofiler/core/ui/editor/analysis/ColumnAnalysisDetailsPage.java [hunk 1]
```diff
@@ -785,20 +792,52 @@ private ReturnCode checkMdmExecutionEngine() {
...
-     * @return whether have a datePatternFreqIndicator in the "analyzed Columns"
+     * @return whether there are java indicator exist.
      */
     @Override
     protected boolean includeJavaEngineIndicator() {
         for (ModelElementIndicator modelElementIndicator : this.treeViewer.getModelElementIndicator()) {
-            if (modelElementIndicator.containsAny(IndicatorEnum.getJavaIndicatorsEnum())) {
+            if (modelElementIndicator.containsAny(IndicatorEnum.getJavaIndicatorsEnum())
+                    || modelElementIndicator.contains(IndicatorEnum.UserDefinedIndicatorEnum)
+                    && searchJUDI(modelElementIndicator, true)) {
```
Comment (xqliu-talend) [inline main/plugins/org.talend.dataprofiler.core/src/org/talend/dataprofiler/core/ui/editor/analysis/ColumnAnalysisDetailsPage.java]: add brackets to make this expression more easy to understand

File: main/plugins/org.talend.dataprofiler.core/src/org/talend/dataprofiler/core/ui/editor/analysis/ColumnAnalysisDetailsPage.java [hunk 2]
```diff
@@ -272,7 +277,9 @@ public void linkActivated(HyperlinkEvent e) {
     @Override
     protected boolean checkSqlEngineIndicatorExist() {
         for (ModelElementIndicator modelElementIndicator : this.treeViewer.getModelElementIndicator()) {
-            if (modelElementIndicator.contains(IndicatorEnum.SqlPatternMatchingIndicatorEnum)) {
+            if (modelElementIndicator.contains(IndicatorEnum.SqlPatternMatchingIndicatorEnum)
+                    || modelElementIndicator.contains(IndicatorEnum.UserDefinedIndicatorEnum)
+                    && searchJUDI(modelElementIndicator, false)) {
```
Comment (xqliu-talend) [inline main/plugins/org.talend.dataprofiler.core/src/org/talend/dataprofiler/core/ui/editor/analysis/ColumnAnalysisDetailsPage.java]: add brackets to make this expression more easy to understand

Comment (zshen-talend) [inline main/plugins/org.talend.dataprofiler.core/src/org/talend/dataprofiler/core/ui/editor/analysis/ColumnAnalysisDetailsPage.java]: Modify it

File: main/plugins/org.talend.dataprofiler.core/src/org/talend/dataprofiler/core/ui/editor/analysis/ColumnAnalysisDetailsPage.java (hunk 1 above)
Comment (zshen-talend) [inline main/plugins/org.talend.dataprofiler.core/src/org/talend/dataprofiler/core/ui/editor/analysis/ColumnAnalysisDetailsPage.java]: modify it

**Pull request merged**